src/header/value.rs

```diff
@@ -359,6 +360,61 @@ impl From<HeaderName> for HeaderValue {
     }
 }
 
+macro_rules! from_integers {
+    ($($t:ident => $max_len:expr),*) => {$(
+        impl From<$t> for HeaderValue {
+            fn from(num: $t) -> HeaderValue {
+                let mut buf = BytesMut::with_capacity($max_len);
+                let _ = ::itoa::fmt(&mut buf, num);
+                HeaderValue {
+                    inner: buf.freeze(),
+                    is_sensitive: false,
+                }
+            }
+        }
+
+        impl HttpTryFrom<$t> for HeaderValue {
+            type Error = Never;
+
+            #[inline]
+            fn try_from(num: $t) -> Result<Self, Self::Error> {
+                Ok(num.into())
+            }
+        }
+
+    )*};
+}
+
+from_integers! {
+    // integer type => maximum decimal length
+
+    // u8 purposely left off... HeaderValue::from(b'3') could be confusing
+    u16 => 5,
+    i16 => 6,
+    u32 => 10,
+    i32 => 11,
+    u64 => 20,
```

Comment:
As you mentioned, this would allocate on 32bit systems. In these cases, you could probably compute the real length using the number value.